fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -444,28 +454,42 @@ public String getPartitionFieldNameInLucene() {
      *
      * @param newRecord record to be written
      * @param groupingKey grouping key
+     * @param assignedPartitionIdHint assigned partition override, if not null
```

Comment:
`hint` seems misleading, it looks like this is used no matter what, replacing any attempt to lookup what it should be.